scope.go

```diff
@@ -947,6 +947,9 @@ func (scope *Scope) typeName() string {
 
 // trace print sql log
 func (scope *Scope) trace(t time.Time) {
+	scope.Set("gorm:trace-time", t)
```

Comment:
IMO, this is a sign of hacky approach. The more or less right way would be to introduce lifecycle hooks, like `BeforeX` and `AfterX` callbacks.